flow/aggregate.py

```diff
@@ -0,0 +1,215 @@
+                            'the aggregate instance. Expected a callable, '
+                            'got {}.'.format(type(func)))
+
+
+class MakeAggregate(Aggregate):
+    r"""This class handles the creation of aggregates.
+
+    .. note::
+        This class should not be instantiated by users directly.
+    :param \*args:
+        Passed to the constructor of :py:class:`Aggregate`.
+    """
+    def __init__(self, *args):
+        super(MakeAggregate, self).__init__(*args)
+
+    def __call__(self, obj, group_name='unknown-operation', project=None):
+        "Return aggregated jobs"
+        aggregated_jobs = list(obj)
+        if self._select is not None:
+            aggregated_jobs = list(filter(self._select, aggregated_jobs))
+        if self._sort is not None:
+            aggregated_jobs = list(sorted(aggregated_jobs,
+                                          key=lambda job: job.sp[self._sort],
+                                          reverse=bool(self._reverse)))
+
+        aggregated_jobs = self._aggregator([job for job in aggregated_jobs])
+        aggregated_jobs = self._create_nested_aggregate_list(aggregated_jobs, group_name, project)
+        if not len(aggregated_jobs):
+            return []
+        return aggregated_jobs
+
+    def _create_nested_aggregate_list(self, aggregated_jobs, group_name, project):
+        # This method converts the returned subset of jobs as an Iterable
+        # from an aggregator function to a subset of jobs as list.
+        aggregated_jobs = list(aggregated_jobs)
+        nested_aggregates = []
+
+        desc = f"Collecting aggregates for {group_name}"
+        for aggregate in tqdm(aggregated_jobs, total=len(aggregated_jobs),
+                              desc=desc, leave=False):
+            try:
+                filter_aggregate = []
+                for job in aggregate:
+                    if job is None:
+                        continue
+                    if project is not None:
+                        if job not in project:
+                            raise ValueError(f'The signac job {str(job)} not found in {project}')
+                    filter_aggregate.append(job)
+                filter_aggregate = tuple(filter_aggregate)
+                if project is not None:
+                    project._aggregates_ids[get_aggregate_id(filter_aggregate)] = \
+                        filter_aggregate
+                nested_aggregates.append(filter_aggregate)
+            except Exception:
+                raise ValueError("Invalid aggregator function provided by "
+                                 "the user.")
+        return nested_aggregates
+
+
+def get_aggregate_id(jobs):
+    """"Generate hashed id for an aggregate of jobs
+
+    :param jobs:
+        The signac job handles
+    :type jobs:
+        tuple
+    """
+    if len(jobs) == 1:
+        return str(jobs[0])  # Return job id as it's already unique
+
+    blob = ''.join((job.get_id() for job in jobs))
```

Comment:
I'd prefer to separate the job ids with a comma for conceptual clarity. I know this isn't really "meaningful" since it's hashed, but I think it makes the purpose clearer.
```suggestion
    blob = ','.join((job.get_id() for job in jobs))
```